Microsoft.DotNet.Maestro.Tasks.Tests/AddAssetTests.cs

```diff
@@ -0,0 +1,64 @@
+using FluentAssertions;
+using Microsoft.DotNet.Maestro.Client.Models;
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+
+namespace Microsoft.DotNet.Maestro.Tasks.Tests
+{
+    [TestFixture]
+    public class AddAssetTests
+    {
+        [Test]
+        public void EmptyAssetList_NewAssetIsOnlyAssetInList()
+        {
+            PushMetadataToBuildAssetRegistry pushMetadata = new PushMetadataToBuildAssetRegistry();
+            List<AssetData> assetData = new List<AssetData>();
+            AssetData expectedAssetData = new AssetData(true) { Name = "testName", Version = "12345", Locations = ImmutableList<AssetLocationData>.Empty.Add(new AssetLocationData(LocationType.None) { Location = "testLocation" }) };
+
+            pushMetadata.AddAsset(assetData, expectedAssetData.Name, expectedAssetData.Version, "testLocation", LocationType.None, true);
+            assetData.Count.Should().Be(1);
+            CompareAssetData(assetData[0], expectedAssetData);
+        }
+
+        [Test]
+        public void AssetsAddedToList_NewAndPreviousAssetsInList()
+        {
+            AssetData existingAssetData = new AssetData(false)
+            {
+                Name = "ExistingAssetName",
+                Version = "56789",
+                Locations = ImmutableList<AssetLocationData>.Empty.Add(new AssetLocationData(LocationType.Container) { Location = "oldTestLocation" })
+            };
+
+            PushMetadataToBuildAssetRegistry pushMetadata = new PushMetadataToBuildAssetRegistry();
+            List<AssetData> assetData = new List<AssetData>() { existingAssetData };
+
+            AssetData newAssetData = new AssetData(true) { Name = "testName", Version = "12345", Locations = ImmutableList<AssetLocationData>.Empty.Add(new AssetLocationData(LocationType.None) { Location = "testLocation" }) };
+
+            pushMetadata.AddAsset(assetData, newAssetData.Name, newAssetData.Version, "testLocation", LocationType.None, true);
+            assetData.Count.Should().Be(2);
+            CompareAssetData(assetData[0], existingAssetData);
+            CompareAssetData(assetData[1], newAssetData);
+        }
+
+        [Test]
+        public void NullAssetList_ThrowsNullReferenceException()
```

Comment:
In your test doc you call out  the following:

Given null parameters (if allowed by caller)

Are the rest of the parameters allowed to be null?

Reply:
See comments below - they can be null and they're just handled as strings the rest of way through the code so I don't think those cases would add much value.